### PR TITLE
fix(1747): adding commit authors to hook schema

### PR DESCRIPTION
**Pull request by klu909**
## Context

Currently there's a mismatch in v3 and v4 ci skip logic. To change v4 ci skip logic to match v3 we need an additional field.

## Objective

This PR add an optional field to hook schema

## References

https://github.com/screwdriver-cd/screwdriver/issues/1747

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by tkyi** [comment]
Can you add a test?

File: test/core/scm.test.js [hunk 1]
```diff
@@ -64,6 +64,10 @@ describe('scm core', () => {
             assert.isNull(validate('scm.hook.push.yaml', core.scm.hook).error);
         });
 
+        it('fails the push hook', () => {
+            assert.isNotNull(validate('scm.hook.push.commiters.yaml', core.scm.hook).error);
```
Comment (jithin1987) [inline test/core/scm.test.js]: can you call this file `cm.hook.push.invalid-commiters.yaml`

File: core/scm.js [hunk 2]
```diff
@@ -192,6 +192,10 @@ const SCHEMA_HOOK = Joi.object().keys({
 
     username: Joi.reach(SCHEMA_USER, 'username'),
 
+    commitAuthors: Joi.array()
```
Comment (tkyi) [inline core/scm.js]: I think it'd be safer to specify what kind of things can be in the array, like https://github.com/screwdriver-cd/data-schema/blob/master/models/template.js#L15

Comment (d2lam) [inline core/scm.js]: I think this should be `Joi.array().items(Joi.string())`

Comment (klu909) [reply core/scm.js]: change done, is there something else that I should change?

Comment (klu909) [reply core/scm.js]: Thanks for info, I have changed it

**Pull request merged**